tracee/tracee.go

```diff
@@ -354,17 +355,24 @@ func (t *Tracee) initBPF(ebpfProgram string) error {
 		return fmt.Errorf("error initializing file_writes perf map: %v", err)
 	}
 
+	// To avoid blocking of processEvents or processFileWrites, buffer error channel
+	t.errorsChannel = make(chan error, 2)
+
 	return nil
 }
 
 // Run starts the trace. it will run until interrupted
 func (t *Tracee) Run() error {
 	sig := make(chan os.Signal, 1)
 	signal.Notify(sig, os.Interrupt)
+	done := make(chan struct{})
+	// Signal pipeline that Tracee exits by closing the done channel
+	defer close(done)
 	t.printer.Preamble()
 	t.eventsPerfMap.Start()
 	t.fileWrPerfMap.Start()
-	go t.processEvents()
+	go t.processErrors()
```

Comment:
why isn't `done` passed to all the gourutines?

Reply:
Done is used by all the go routines in the pipeline. It is used to avoid one stage in the pipeline to block the stages which come after it. Which go routines do you refer to?

Reply:
the one I commented about, for example `processErrors` (it's outside the pipeline)

Reply:
Well, we can add it there too, but it's not a must (it will not block anything like in the pipeline)